stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/state/ChannelMutableState.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.client.channel.state
+
+import io.getstream.chat.android.client.events.ChatEvent
+import io.getstream.chat.android.client.models.ChannelData
+import io.getstream.chat.android.client.models.ChannelUserRead
+import io.getstream.chat.android.client.models.Config
+import io.getstream.chat.android.client.models.Member
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import java.util.Date
+
+@Suppress("VariableNaming")
+public interface ChannelMutableState : ChannelState {
+
+    public val _messages: MutableStateFlow<Map<String, Message>>
+    public val _watcherCount: MutableStateFlow<Int>
+    public val _typing: MutableStateFlow<Map<String, ChatEvent>>
+    public val _reads: MutableStateFlow<Map<String, ChannelUserRead>>
+    public val _read: MutableStateFlow<ChannelUserRead?>
+    public val _endOfNewerMessages: MutableStateFlow<Boolean>
+    public val _endOfOlderMessages: MutableStateFlow<Boolean>
+    public val _loading: MutableStateFlow<Boolean>
+    public val _hidden: MutableStateFlow<Boolean>
+    public val _muted: MutableStateFlow<Boolean>
+    public val _watchers: MutableStateFlow<Map<String, User>>
+    public val _members: MutableStateFlow<Map<String, Member>>
+    public val _loadingOlderMessages: MutableStateFlow<Boolean>
+    public val _loadingNewerMessages: MutableStateFlow<Boolean>
+    public val _channelData: MutableStateFlow<ChannelData?>
+    public val _oldMessages: MutableStateFlow<Map<String, Message>>
+    public val lastMessageAt: MutableStateFlow<Date?>
+    public val _repliedMessage: MutableStateFlow<Message?>
+    public val _unreadCount: MutableStateFlow<Int>
+    public val _membersCount: MutableStateFlow<Int>
+    public val _insideSearch: MutableStateFlow<Boolean>
+
+    /** Channel config data. */
+    public val _channelConfig: MutableStateFlow<Config>
+
```

Comment:
We can keep this as an internal interface and move and change visibility later, as we discussed. With an internal interface keeping the original names can be good to avoid increasing `ChannelMutableStateImpl` now. What do you think?

Reply:
We've already done such a thing in `MutableGlobalState`.
Those properties are meant to be private. Even if those are for internal SDK usage I would avoid providing full control to other parts of the SDK by exposing a `MutableStateFlow`.

Reply:
I took a look at the code and this would be no trivial task because of the implementation details of `ChannelMutableStateImpl`. One example would be the removal of messages: 
```
override fun removeLocalMessage(message: Message) {
        mutableState.setMessages(mutableState.messages.value.associateBy { it.id } - message.id)
    }
```

Every time it would be an association, which would impact the performance, because ChannelState exposes messages as a `List<Message>`, but the implementation uses a `Map<String, Message>`. Let's create a ticket just for that, it can be done right after this PR.

Reply:
🤔  Although I can change the other fields, and leave the ones with this map/list relation